docs/INTERNAL_DOCKER_IMAGES.md

```diff
@@ -0,0 +1,81 @@
+# Internal Docker Images Publish Process
```

Comment:
@elastic/observablt-ci 

I was unsure about the naming and location of this file. WDYT?